### DISCUSSION TITLE
Only local connections work for me.

### DISCUSSION THREAD

**Discussion by X7895** [jchristn/SuperSimpleTcp#191] [General]
Local connections using this works fine:

server1 = new SimpleTcpServer("127.0.0.1:9000");

Client1 = new SimpleTcpClient("127.0.0.1:9000");

External connections using this code don't work:

IP of the PC that hosts the server: 67.231.135.128

server1 = new SimpleTcpServer("0.0.0.0:9000");

Client1 = new SimpleTcpClient("67.231.135.128:9000");

Is this code correct?
Windows Firewalls are disabled.
What else could be blocking the connection?

**Comment by jchristn** [comment]
Hi, are you running your server as administrator?

**Reply by X7895** [reply]
Yes.
It doesn't work either.

**Reply by jchristn** [reply]
Are there network firewalls anywhere in the picture?  Are you deployed on machines you control or on EC2 or other cloud service?

**Reply by X7895** [reply]
Just 2 normal PCs with Windows 10. One starts the serverprogram the other one the client program.

**Reply by jchristn** [reply]
Are they both on the same network?  You're attempting to connect to a 67.x address but I'm guessing that's your public IP, and the machines are on the same private network.  Is that the case?

**Reply by X7895** [reply]
No they are not on the same private network.
Its 2 PCs in different Networks that have to connect over the internet.
They both have different public IPs.
The 67.231.135.128 adress was just an example for a public IP. 
I of course use my real public IP adress for the server.

**Reply by jchristn** [reply]
Is that IP actually assigned to the machine or is it assigned to a NAT router?  Can you send me the output of ```ipconfig```?  Also are these laptops/PCs that you have physically, or are they virtual machines?  And if virtual machines, are they on a machine you control or a cloud instance?

**Reply by X7895** [reply]
Its 2 physical PCs in 2 different houses.
The IP is the public IP of the NAT router the PC that hosts the server is behind.

Ipconfig:
Verbindungsspezifisches DNS-Suffix: home
   IPv6-Adresse. . . . . . . . . . . : 2a02:908:950:e6e0:def9:ba3a:f31c:e3ce
   Temporäre IPv6-Adresse. . . . . . : 2a02:908:950:e6e0:b123:c3e6:a3db:19dd
   Verbindungslokale IPv6-Adresse  . : fe80::130b:4dc2:47ee:fe97%2
   IPv4-Adresse  . . . . . . . . . . : 192.168.0.122
   Subnetzmaske  . . . . . . . . . . : 255.255.255.0
   Standardgateway . . . . . . . . . : fe80::925c:44ff:fecf:921b%5
                                       192.168.0.1

**Comment by jchristn** [comment]
Ok.  Your listener then has to be on ```0.0.0.0``` and your router must be set up to port-forward inbound requests on port 9000 to 192.168.0.122.
